src/ucp/wireup/address.c

```diff
@@ -1653,6 +1602,25 @@ static ucp_rsc_index_t ucp_address_get_remote_device_index(
     return ucs_array_length(device_array) - 1;
 }
 
+static void
+ucp_address_unpacked_adjust_legacy(ucp_unpacked_address_t *unpacked_address)
+{
+    uint8_t legacy_minor_api_ver = ucp_address_packed_release_version(18);
+    size_t i;
+
+    if ((unpacked_address->addr_version != UCP_OBJECT_VERSION_V1) ||
+        (unpacked_address->dst_version >
+         ucp_address_unpack_release_version(legacy_minor_api_ver))) {
+        return;
+    }
+
+    /* reset legacy bits in md_index */
+    for (i = 0; i < unpacked_address->address_count; ++i) {
+        unpacked_address->address_list[i].md_index &=
+                ~UCP_ADDRESS_FLAG_MD_INDEX_LEGACY_BITS;
```

Comment:
seems we still have a mask for md index based on the address version. maybe just calculate the mask and use it when unpacking the md index, instead of adjusting the md_index later? this way the md_index will never be in a wrong intermediate state, and less changes vs. previous code

Reply:
in addr v1 UCX version is packed in the end of address, so we cannot calculate the mask in advance